### PR TITLE
SE - Property reference: Test case for auto properties are not learning constraints

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Test case for #6930

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs [hunk 1]
```diff
@@ -781,6 +781,47 @@ public void PropertyReference_Write_SetsNotNull()
             validator.ValidateTag("AfterIndexer", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue());
         }
 
+        [TestMethod]
+        public void PropertyReference_Write_SetsNull_AndRead_SetsNotNull_ForAutoProperties()
+        {
+            const string code = """
+                public object AutoProperty { get; set; }
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:788]: What about moving this into `SETestContext` with the others, like `Property`?

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs [hunk 2]
```diff
@@ -781,6 +781,47 @@ public void PropertyReference_Write_SetsNotNull()
...
+                }
+                """;
+            var validator = SETestContext.CreateCSMethod(code).Validator;
+            validator.ValidateContainsOperation(OperationKind.PropertyReference);
+            validator.ValidateTag("AfterSetNull", x => x.Should().HaveNoConstraints()); // FN Auto-properties should behave the same as fields. Expected: HaveOnlyConstraint(ObjectConstraint.Null)
+            validator.ValidateTag("AfterReadReference", x => x.Should().HaveNoConstraints()); // FN Expected: HaveOnlyConstraint(ObjectConstraint.NotNull)
+        }
+
+        [TestMethod]
+        public void PropertyReference_NotTrackedFor_FullProperties()
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:805]: To explain the context and then say the expectations
```suggestion
        public void PropertyReference_FullProperties_NotTracked()
```

**Review by pavel-mikula-sonarsource — CHANGES_REQUESTED** [review]
We'll eventually need tests for `IOperationExtensionsTest`. Are those going to be part of the implementation later?

File: analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs [hunk 3]
```diff
@@ -781,6 +781,47 @@ public void PropertyReference_Write_SetsNotNull()
...
+                {
+                    AutoProperty = null;
+                    Tag("AfterSetNull", AutoProperty);
+                    AutoProperty.ToString();
+                    Tag("AfterReadReference", AutoProperty);
+                }
+                """;
+            var validator = SETestContext.CreateCSMethod(code).Validator;
+            validator.ValidateContainsOperation(OperationKind.PropertyReference);
+            validator.ValidateTag("AfterSetNull", x => x.Should().HaveNoConstraints()); // FN Auto-properties should behave the same as fields. Expected: HaveOnlyConstraint(ObjectConstraint.Null)
```
[the remark this answers is not in the document]
Comment (Tim-Pohlmann) [reply analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs:800]: Education: What is the difference between FN and FIXME in this context?

[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs]: @Tim-Pohlmann 
Technically it's not FN, as there are no issues involved related to any rule. It's just incorrect behavior of some part of the code.

`FIXME` should never make it to `master` and must be addressed before the feature branch is merged. It is tracked by S1134.
`ToDo` could make it to master and be done later. It is tracked by S1135.

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**